File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numSet = set(nums)
        longest = 0  # initially
        for num in numSet:
            if (num - 1) not in numSet:
                length = 1  # initially
                while (num + length) in numSet:
                    length += 1
                longest = max(length, longest)
        return longest

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function (nums) {
    if (nums.length === 0) return 0;
    const set = new Set(nums);

    const seqStart = [];
    set.forEach((num) => {
        if (!set.has(num - 1)) seqStart.push(num);
    });

    const counts = [];
    seqStart.forEach((num) => {
        let count = 1;
        while (set.has(num + 1)) {
            count += 1;
            num++;
        }
        counts.push(count);
    });

    return Math.max(...counts);

};